lib/js-yaml/dumper.js

```diff
@@ -262,7 +251,9 @@ function chooseScalarStyle(string, singleLineOnly, indentPerLevel, lineWidth, te
   var shouldTrackWidth = lineWidth !== -1;
   var previousLineBreak = -1; // count the first line correctly
   var plain = isPlainSafeFirst(string.charCodeAt(0))
-          && !isWhitespace(string.charCodeAt(string.length - 1));
+          && !isWhitespace(string.charCodeAt(string.length - 1))
+          && /\s#/.test(string) === false
+          && /:(?:\s|$)/.test(string) === false;
```

Comment:
All checks should be done via `.charCodeAt()` for perfomance reasons.

Also, this lines are not descriptive - difficult to understand.

Reply:
Would comments help (apart from the fact that these are otherwise undesirable, see also my comments further below about the new insights about this)?

Reply:
Yes, comments help in such cases. Or descriptive helper function names (like `isWhiteSpace`, `hasCommentPattern`)

But regexps should not be used - see any other part of code, there are no regexps.